app/src/main/java/to/dev/dev_android/view/main/view/CustomWebChromeClient.kt

```diff
@@ -21,6 +24,11 @@ class CustomWebChromeClient(val context: Context,
         return true
     }
 
+    override fun onProgressChanged(view: WebView, newProgress: Int) {
+        super.onProgressChanged(view, newProgress)
+        if (newProgress == 100 && view.url == context.getString(R.string.base_url)) view.clearHistory()
```

Comment:
I get the need of compare the url but having an instance of the context just for fetching the URL seems over working and can lead to memory leaks. 

I'd suggest if this matters to _clear the back history_ we can pass this value directly in the constructor instead of the `context`:

```kotlin
class CustomWebChromeClient(
    val baseURL: String,
    val binding: ActivityMainBinding,
    private val listener: CustomListener
) 
```

No other parts of the class are using the `context` anyway

Reply:
I'm noticing that this `base_url` is also set in the `build.gradle` of the module `baseui` so we can access this same url from `BuildConfig.baseUrl` (which seems kind of wrong but we remove the need for the `Context`)

Reply:
Hey @sierisimo! Thanks for your comments, yeah I believe we can get rid of the context, I just wonder which approach is the best...
@maestromac any preference?

Reply:
Seems it's up to you. I will suggest: pass it over the constructor. Even when it's available over the `BuildConfig` we can make the client independent with the constructor being the one set.

Reply:
Best to create an environment config class that abstracts retrieving such data, such as the base url. Also string resources shouldn't be used for urls